### PR TITLE
Language feature fixed

### LINKED ISSUE #1685: The language feature is not working
**Actual Behaviour**

No matter what language you select, the text language stays english.

**Expected Behaviour**

The langauge should change depending on the user preference.

**Steps to reproduce it**

Change the language settings in the settings option and click save.

**Screenshots of the issue**
The language I had selected was *Hindi*, but neither the menu nor the text on the pages changed.
screenshot-chat susi ai-2018 11 05-02-04-34
screenshot-chat susi ai-2018 11 05-02-04-21

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #1685 

Changes: I have fixed the language feature, and provided with a more efficient implementation. I have also made it so that in case a phrase is not available in the selected language, the system falls back to English for that particular phrase.

Demo Link: https://pr-1686-fossasia-susi-web-chat.surge.sh

Screenshots for the change: 
screenshot-localhost-3000-2018 11 05-02-18-04
screenshot-localhost-3000-2018 11 05-02-17-55

screenshot-localhost-3000-2018 11 05-02-17-04
screenshot-localhost-3000-2018 11 05-02-16-38

### CODE REVIEW

**Comment by AakashMallik** [comment]
@rajvaibhavdubey There is a peculiar codacy issue. Could you help me out with this?

File: src/components/Translate/Translate.react.js [hunk 1]
```diff
@@ -14,22 +14,8 @@ import nl from './nl.json';
...
-  }
-
-  changeLanguage = text => {
-    this.setState({
-      text: text,
-    });
-  };
-
-  lang = {
+const Translate = props => {
```
Comment (akshatnitd) [inline src/components/Translate/Translate.react.js]: Use `(props)`

File: src/components/Translate/Translate.react.js [hunk 2]
```diff
@@ -43,31 +29,23 @@ class Translate extends Component {
     'nl-NL': nl,
   };
 
-  componentDidMount() {
-    let arr,
-      file,
-      defaultPrefLanguage = this.state.defaultPrefLanguage;
-    let text = this.state.text;
+  const getTranslation = text => {
```
Comment (akshatnitd) [inline src/components/Translate/Translate.react.js]: Use `(text)`

File: src/components/Translate/Translate.react.js [hunk 3]
```diff
@@ -43,31 +29,23 @@ class Translate extends Component {
...
+    const defaultPrefLanguage = UserPreferencesStore.getPrefLang();
+    let translatedText = undefined;
```
Comment (akshatnitd) [inline src/components/Translate/Translate.react.js]: Use null instead

File: src/components/Translate/Translate.react.js (hunk 1 above)
Comment (AakashMallik) [reply src/components/Translate/Translate.react.js]: Sir, Eslint is changing is back to props as soon as i use git commit

File: src/components/Translate/Translate.react.js (hunk 2 above)
Comment (AakashMallik) [reply src/components/Translate/Translate.react.js]: same as above

**Comment by AakashMallik** [comment]
I have made the requested changes, sir.

File: src/components/Translate/Translate.react.js [hunk 4]
```diff
@@ -43,31 +29,23 @@ class Translate extends Component {
...
     if (defaultPrefLanguage !== 'en-US') {
-      if (this.lang.hasOwnProperty(defaultPrefLanguage)) {
-        arr = Object.keys(this.lang[defaultPrefLanguage]);
-        file = this.lang[defaultPrefLanguage];
+      if (lang.hasOwnProperty(defaultPrefLanguage)) {
+        translatedText = lang[defaultPrefLanguage][text];
       } else {
-        arr = Object.keys(jp);
-        file = jp;
+        translatedText = jp[text];
```
Comment (akshatnitd) [inline src/components/Translate/Translate.react.js]: Does this mean, we default to japanese?

Comment (AakashMallik) [reply src/components/Translate/Translate.react.js]: @akshatnitd Yes sir, at least that is what the previous developer intended to do, so I just made it functional. If you want I can change it to English.

Comment (akshatnitd) [reply src/components/Translate/Translate.react.js]: Yes, that would be right

Comment (AakashMallik) [reply src/components/Translate/Translate.react.js]: @akshatnitd Made the changes you asked for. Thanks for your time, sir.

**Pull request merged**